glide-core/redis-rs/redis/src/cmd.rs

```diff
@@ -30,6 +31,8 @@ pub struct Cmd {
     cursor: Option<u64>,
     // If it's true command's response won't be read from socket. Useful for Pub/Sub.
     no_response: bool,
+    /// The span associated with this command
+    span: Option<GlideSpan>,
```

Comment:
I think we have some mismas with the definition of span and events and trace. 
A trace is like a "path" that describes the journey of a request or an action through your system.
A span is a single step or operation within that journey.
If the trace represents the full path, the spans are the individual stops along the way.

For example:
```
A request reaches the API Gateway (Span 1).
The request is passed to a service for processing (Span 2).
The service queries the database (Span 3).
The result is returned to the user (Span 4).
```
is mean, the first span in the journey is often called the Root Span, representing the start of the trace. All other spans are its children. So, in this sturct the span should called **trace or root_span**.

Span event dont have start and end time, it has only a singular point in time. see https://opentelemetry.io/docs/concepts/signals/traces/#span-events

if so, we need also to add a function to return all spans in a trace.